pkg/system/phase2_creating.go

```diff
@@ -713,59 +715,88 @@ func (r *Reconciler) SetDesiredAgentProfile(profileString string) string {
 	return string(profileBytes)
 }
 
+func (r *Reconciler) setKMSConditionStatus(s corev1.ConditionStatus) {
+	conditions := &r.NooBaa.Status.Conditions
+	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
+		LastHeartbeatTime: metav1.NewTime(time.Now()),
+		Type:              nbv1.ConditionTypeKMS,
+		Status:            s,
+	})
+	r.Logger.Infof("setKMSConditionStatus %v", s)
+}
+
+
+// initExternalKMS runs when no Secret Data found for Secret ID
+func (r *Reconciler) initExternalKMS(c secrets.Secrets, secretPath, keySecretName string) error {
```

Comment:
initExternalKMS is not a good function name, it sounds like you are initing the connection to the KMS and not putting the secret in it...

Reply:
My line of thought is the KMS storage is uninitialized before the initial key is written. How would you like to call it, @romayalon?

Reply:
initKeyInKMS or something similar, I feel that we need that key keyword there

Reply:
👍 See [commit](https://github.com/noobaa/noobaa-operator/pull/797/commits/fac3ea169cc6f5448d94e07f5439f86ecebee81c)